honeybadger/contrib/aws_lambda.py

```diff
@@ -0,0 +1,131 @@
+import sys
+
+from honeybadger import honeybadger
+from honeybadger.plugins import Plugin
+from honeybadger.utils import filter_dict
+
+from threading import local
+
+_thread_locals = local()
+REQUEST_LOCAL_KEY = '__awslambda_current_request'
+
+def get_execution_params():
+    """
+    Return current execution event & context for this thread.
+    """
+    return getattr(_thread_locals, REQUEST_LOCAL_KEY, None)
+
+def set_execution_params(aws_event, aws_context):
+    """
+    Set current execution event & context for this thread.
+    """
+    request = {
+        'aws_event':aws_event,
+        'aws_context':aws_context
+    }
+    setattr(_thread_locals, REQUEST_LOCAL_KEY, request)
+
+def clear_execution_params():
+    """
+    Clears execution params for this thread.
+    """
+    if hasattr(_thread_locals, REQUEST_LOCAL_KEY):
+        setattr(_thread_locals, REQUEST_LOCAL_KEY, None)
+
+def reraise(tp, value, tb=None):
+    """
+    Re-raises a caught error
+    """
+    assert value is not None
+    if value.__traceback__ is not tb:
+        raise value.with_traceback(tb)
+    raise value
+    
+def get_lambda_bootstrap():
+    """
+    Get AWS Lambda bootstrap module
+    """
+
+    if "bootstrap" in sys.modules:
+        return sys.modules["bootstrap"]
+    elif "__main__" in sys.modules:
+        if hasattr(sys.modules["__main__"], "bootstrap"):
+            return sys.modules["__main__"].bootstrap  # type: ignore
+        return sys.modules["__main__"]
+    else:
+        return None
+
+def _wrap_lambda_handler(handler):
+
+    def wrapped_handler(aws_event, aws_context, *args, **kwargs):
+        set_execution_params(aws_event, aws_context)
+
+        honeybadger.begin_request(aws_event)
+        try:
+            return handler(aws_event, aws_context, *args, **kwargs)
+        except Exception as e:
+            honeybadger.notify(e)
+            exc_info = sys.exc_info()
+            clear_execution_params()
+            honeybadger.reset_context()
+
+            #Rerase exception to proceed with normal aws error handling
+            reraise(*exc_info)
+
+    return wrapped_handler
+
+
+class AWSLambdaPlugin(Plugin):
+
+    def __init__(self):
+        super(AWSLambdaPlugin, self).__init__('AWSLambda')
+        lambda_bootstrap = get_lambda_bootstrap()
+        self.initialize_request_handler(lambda_bootstrap)
+        
+    def supports(self, config, context):
+        return config.is_aws_lambda_environment
+
+    def generate_payload(self, config, context):
+        """
+        Generate payload by checking the lambda's
+        request event & context.
+        """
+        request = get_execution_params()
+        event = request['aws_event']
+        context = request['aws_context']
+
+        payload = {
```

Comment:
@Kelvin4664 I'm liking the plugin approach.

Is this supposed to be [a `notice["request"]` payload](https://github.com/honeybadger-io/honeybadger-schemas/blob/master/exceptions/v1/notices.json#L258)? It looks like it is in [the Django plugin](https://github.com/honeybadger-io/honeybadger-python/blob/master/honeybadger/contrib/django.py#L71).

You'll need to make sure that the context is properly showing up in the Honeybadger UI here:

![image](https://user-images.githubusercontent.com/474649/108397425-cbb78080-71cc-11eb-9c03-6951aeaf57f5.png)